Removed 'canvas' property from 'IWebGL2RenderingContext'

I tried building `xterm.js` recently, and got the following error:

```
addons/xterm-addon-webgl/src/GlyphRenderer.ts:305:26 - error TS2339: Property 'canvas' does not exist on type 'IWebGL2RenderingContext'.

305     gl.viewport(0, 0, gl.canvas.width, gl.canvas.height);   
```

Looks like the [`canvas`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/canvas) property was removed as per https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029, but was that intentional? MSDN says

> WebGL 2 is an extension to WebGL 1. The WebGL2RenderingContext interface implements all members of the WebGLRenderingContext interface. Some methods of the WebGL 1 context can accept additional values when used in a WebGL 2 context. You will find this info noted on the WebGL 1 reference pages.

--- a/baselines/dom.generated.d.ts
+++ b/baselines/dom.generated.d.ts
@@ -16482,7 +16482,7 @@ declare var WebGLRenderingContext: {
 };
 
 interface WebGLRenderingContextBase {
-    canvas: HTMLCanvasElement;
+    readonly canvas: HTMLCanvasElement;
     readonly drawingBufferHeight: GLsizei;
     readonly drawingBufferWidth: GLsizei;
     activeTexture(texture: GLenum): void;

--- a/baselines/dom.generated.d.ts
+++ b/baselines/dom.generated.d.ts
@@ -16482,6 +16482,7 @@ declare var WebGLRenderingContext: {
 };
 
 interface WebGLRenderingContextBase {
+    canvas: HTMLCanvasElement;
     readonly drawingBufferHeight: GLsizei;
     readonly drawingBufferWidth: GLsizei;
     activeTexture(texture: GLenum): void;

--- a/inputfiles/addedTypes.json
+++ b/inputfiles/addedTypes.json
@@ -179,6 +179,17 @@
                     ]
                 }
             },
+            "WebGLRenderingContextBase": {
+                "properties": {
+                    "property": {
+                        "canvas": {
+                            "name": "canvas",
+                            "exposed": "Window",
+                            "type": "HTMLCanvasElement"
+                        }
+                    }
+                }
+            },
             "WindowEventHandlers": {
                 "events": {
                     "event": [

--- a/inputfiles/addedTypes.json
+++ b/inputfiles/addedTypes.json
@@ -183,9 +183,12 @@
                 "properties": {
                     "property": {
                         "canvas": {
-                            "name": "canvas",
                             "exposed": "Window",
-                            "type": "HTMLCanvasElement"
+
+                            // https://github.com/KhronosGroup/WebGL/pull/3296 will make these redundant
+                            "name": "canvas",
+                            "type": "HTMLCanvasElement",
+                            "readonly": true
                         }
                     }
                 }

--- a/inputfiles/removedTypes.json
+++ b/inputfiles/removedTypes.json
@@ -16,6 +16,7 @@
             "WebGLRenderingContextBase": {
                 "properties": {
                     "namesakes": {
+                        // This will become redundant when https://github.com/KhronosGroup/WebGL/pull/3296 merges.
                         "canvas": null
                     }
                 }

--- a/inputfiles/removedTypes.json
+++ b/inputfiles/removedTypes.json
@@ -16,7 +16,7 @@
             "WebGLRenderingContextBase": {
                 "properties": {
                     "namesakes": {
-                        // This will become redundant when https://github.com/KhronosGroup/WebGL/pull/3296 merges.
+                        // https://github.com/KhronosGroup/WebGL/pull/3296 will make this redundant
                         "canvas": null
                     }
                 }

